packages/api/src/shared/analytics.ts

```diff
@@ -18,14 +19,15 @@ export interface EventMessageV1 extends IdentifyMessageV1 {
 
 const postApiKey = Config.getPostHogApiKey();
 
-export const analytics = (params: EventMessageV1) => {
+export const analytics = (params: EventMessageV1 & { apiType: Product | "internal" }) => {
```

Comment:
Should we just pass all params as properties? For example distintId is not within properties in the params. Which means were doing nothing with that variable (which we should - i added to the current PR im working on but could be done here).

Reply:
TBH I think we should review how we do event analytics, Posthog has a recommended approach, we could adjust the code to work like that and, for each component of the event name we could also add a respective custom property.

And what you said, we need to require some params, otherwise we get empty and can't filter it.

OTOH, I don't think we can pass all params as properties, as some events might have properties that others don't.

With that being said, we could try to abstract a bit some stuff, e.g.:
- we have URL and action, and it seems only one of them is set on each event (like, action is usually for internal actions and/or UI and URL is for the external requests)... maybe we could have something that we could use as the context for both?

Reply:
https://github.com/metriport/metriport-internal/issues/1421